package.json

```diff
@@ -45,14 +45,15 @@
     "ssb-bfe": "^3.7.0",
     "ssb-box2": "^4.0.0",
     "ssb-crut": "^4.6.1",
-    "ssb-db2": "^6.2.6",
-    "ssb-meta-feeds": "^0.38.1",
+    "ssb-db2": "6.3.0",
+    "ssb-meta-feeds": "0.38.1",
```

Comment:
Upgrading to 0.38.2 breaks tests

```
get added to a group
    ✔ tribes2 started for both alice and bob
    ✔ alice and bob replicate their trees
    in empty function
    emptyFeed undefined
    about to post root msg
    just published root msg
    ✔ alice created a group
    ✔ alice added bob to the group
    ✖ 5) Error: retryUntil timed out
      Expected error to be falsy
      

  add member
```

Reply:
Sometimes it fails in this place, sometimes in the next test or later. It is really flaky.

@arj03 do you know what might be up here? this is the pr for .2 https://github.com/ssbc/ssb-meta-feeds/pull/112/files?diff=split&w=1

I tried to dig and get some more info using this

```
diff --git a/package.json b/package.json
index 3e16c57..86b5c1c 100644
--- a/package.json
+++ b/package.json
@@ -46,7 +46,7 @@
     "ssb-box2": "^4.0.0",
     "ssb-crut": "^4.6.1",
     "ssb-db2": "6.3.0",
-    "ssb-meta-feeds": "0.38.1",
+    "ssb-meta-feeds": "0.38.2",
     "ssb-private-group-keys": "^1.1.1",
     "ssb-ref": "^2.16.0",
     "ssb-uri2": "^2.4.1"
diff --git a/test/helpers/replicate.js b/test/helpers/replicate.js
index 2338965..60ff5aa 100644
--- a/test/helpers/replicate.js
+++ b/test/helpers/replicate.js
@@ -4,6 +4,16 @@
 
 const { promisify: p } = require('util')
 const pull = require('pull-stream')
+const paraMap = require('pull-paramap')
+const {
+  where,
+  and,
+  isDecrypted,
+  type,
+  live,
+  author,
+  toPullStream,
+} = require('ssb-db2/operators')
 const deepEqual = require('fast-deep-equal')
 
 /**
@@ -31,6 +41,24 @@ module.exports = async function replicate(person1, person2, opts = {}) {
   let drain1
   pull(
     person1.metafeeds.branchStream({ old: true, live: true }),
+    paraMap((branch, cb) => {
+      if (
+        branch.length === 4 &&
+        branch[3].purpose === 'group/additions' &&
+        branch[3].seed !== null
+      ) {
+        console.log('person1 p1 additions feed', branch[3])
+        return pull(
+          person1.db.query(where(author(branch[3].id)), toPullStream()),
+          pull.collect((err, res) => {
+            console.log('p1 seeing p1 additions', res)
+
+            return cb(null, branch)
+          })
+        )
+      }
+      return cb(null, branch)
+    }),
     pull.flatten(),
     pull.map((feedDetails) => feedDetails.id),
     pull.unique(),
@@ -43,6 +71,24 @@ module.exports = async function replicate(person1, person2, opts = {}) {
   let drain2
   pull(
     person2.metafeeds.branchStream({ old: true, live: true }),
+    paraMap((branch, cb) => {
+      if (
+        branch.length === 4 &&
+        branch[3].purpose === 'group/additions' &&
+        branch[3].seed === null
+      ) {
+        console.log('person2 p1 additions feed', branch[3])
+        return pull(
+          person2.db.query(where(author(branch[3].id)), toPullStream()),
+          pull.collect((err, res) => {
+            console.log('p2 seeing p1 additions', res)
+
+            return cb(null, branch)
+          })
+        )
+      }
+      return cb(null, branch)
+    }),
     pull.flatten(),
     pull.map((feedDetails) => feedDetails.id),
     pull.unique(),
```

I thought that the problem might be that p2 gets p1's additions feed but doesn't manage to decrypt it, but p2 doesn't seem to replicate p1's additions feed even if p2 discovers it. I.e. it prints stuff like

```
p2 seeing p1 additions []
    p1 seeing p1 additions [
      {
        key: '%kcZefmGgRvoiiQgvW6PuipCA71XQf8DANVnJMYo6kX4=.sha256',
        value: {
...
```

-------

hmm although i write that, if i switch back to .1, my small script above still prints [] for p2 and the tests pass, so my script is buggy in some way i guess. gotta go eat though.

Reply:
moved to https://github.com/ssbc/ssb-tribes2/issues/48